cirq/study/resolver.py

```diff
@@ -102,8 +100,8 @@ def value_of(self,
 
         # Input is a symbol (sympy.Symbol('a')) and its string maps to a number
         # in the dictionary ({'a': 1.0}).  Return it.
-        if isinstance(value, sympy.Symbol) and str(value) in self.param_dict:
-            param_value = self.param_dict[str(value)]
+        if (isinstance(value, sympy.Symbol) and value.name in self.param_dict):
```

Comment:
Is it possible for `symbol.name` and `str(symbol)` to ever disagree?

Reply:
I don't think so.  The string printing code in sympy is worthy of a scary programmer-themed Halloween movie, but, as far as I can, this basically just outputs symbol.name, at least for sympy.Symbol types.

https://github.com/sympy/sympy/blob/master/sympy/printing/str.py#L747